src/App.tsx

```diff
@@ -15,36 +15,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = '',
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>([...goodsFromServer]);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sortGoods = (type: SortType) => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (type === SortType.Alphabetically) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    }
+
+    if (type === SortType.Length) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    setSortType(type);
+  };
+
+  const reverseGoods = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(prev => !prev);
+  };
+
+  const resetGoods = () => {
+    setGoods([...goodsFromServer]);
+    setSortType(SortType.None);
+    setIsReversed(false);
+  };
+
   return (
```

Comment:
[UX] - The active button should be marked to indicate to the user what action is active. Currently, the 'is-light' class is used to indicate an inactive state, but there's no clear indication for an active state. Consider adding a distinct style or class for the active state.